File: 2021/1/part1.py
def part1(filename):
	with open(filename, 'r') as file:
		data = [int(l) for l in file.read().splitlines()]

	count = 0

	for i in range(1, len(data)):
		if data[i] > data[i-1]:
			count += 1

	return count

if __name__ == '__main__':
	count = part1('input.txt')
	print(count)

File: 2022/06/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func part1() {
	file, _ := os.Open("input.txt")
	scanner := bufio.NewScanner(file)
	defer file.Close()

	scanner.Scan()
	input := scanner.Text()

	for i := range input[:len(input)-4] {
		distinct := true
		for _, c := range input[i : i+4] {
			if strings.Count(input[i:i+4], string(c)) > 1 {
				distinct = false
				break
			}
		}

		if distinct {
			fmt.Println(i + 4)
			break
		}
	}
}
